### ISSUE TITLE
Detect more than 255 classes

### ISSUE THREAD

**Issue by denis-sumin** [facebookresearch/maskrcnn-benchmark#1169]
## ❓ Questions and Help
Hi,
I experimented with the models in maskrcnn-benchmark successfully with several datasets that consist of a different number of classes. That worked well until the number of classes in a dataset was below 255. When using more than 255 classes, I'm still getting labels in the range of [1..255].

I suspect that I should change the CONV_HEAD_DIM setting:
```MODEL.ROI_BOX_HEAD.CONV_HEAD_DIM = 256```

Is it a correct guess (I'm starting a new train to check that now), are there any other ideas what could clip the classifier?

**Comment by fmassa** [comment]
Hi,

`MODEL.ROI_BOX_HEAD.CONV_HEAD_DIM` should not affect the outcome of the model.
I would expect that you might have a problem loading the labels in your dataset, where the labels are read as `uint8` values, but they should be read as a larger int.

**Comment by denis-sumin** [comment]
Hi, 
Thank you for your reply.

That I have already checked and didn't find any problem with the integer type. Thus I started to think if the network architecture itself could clip the labels (either while reading them or when storing them).
I'm using the BoxList class from the maskrcnn-benchmark which stores labels as a tensor in the `extra_fields`. There is no typecast there. I also traced the `targets` values during the network pipeline and didn't find that they are getting clipped anywhere.

I'm using the simple detection mode (not the segmentation, not the keypoint detection; perhaps, is it Fast mode). `grep`-ing the project with the uint8 keyword doesn't show that the type is used anywhere in the chain.
